### ISSUE TITLE
Language Support besides Typescript

### ISSUE THREAD

**Issue by jhuntoo** [CitadelJS/core#2]
I know @remojansen has mentioned that we should support ES6/7 and ES5 too ??

What are the implications for the project?

**Comment by lholznagel** [comment]
Taking a look on http://node.green/ I would say we should concentrate on ES 6 / 7

**Comment by jhuntoo** [comment]
Agreed, will using ES6/ES7 sacrifice any features that typescript delivers ?

**Comment by lholznagel** [comment]
not that I know. await async has a limitation on es5
But most times typescript has more features than es6. Will make a quick search

**Comment by lholznagel** [comment]
Another limitation is the node version. we would need to use node > 6.x
node 5.x does not support so much es6

**Comment by otbe** [comment]
Decorators could be tricky at all. No node support, different behavior in TS/Babel.

I think Babel Support would be nice too...So users can decide what node version to use. Think of a serverless AWS lambda app. Node is restricted to v0.10.x and v4.x.x for lambdas.

**Comment by remojansen** [comment]
At the moment async/await only works when targeting ES5 in node > 6.x but that won't be a problem soon. TypeScript 2.0 will be released really soon (probably before we are ready to release Citadel) and it will support async/await when targeting ES5. 

I suggest installing `typescript@next` so we can use the 2.0 features, target ES5 and support node 0.10.x.

I also think that supporting Babel is important for those that don't like TS but TypeScript should be our main focus.

**Comment by otbe** [comment]
Did some testing on this. We can use ```es2015```or ```es6``` as target for TS compiler, without have to use babel at all for building citadel.
It will work for node v6 out of the box and for v5 and v4 with ```--harmony``` switch. We have also not to provide polyfills or obscure typings for such things like ```Promise``` or ```Array.prototype.find```.

I think targeting ES5 with ```async/await``` is not in ```typescipt@next``` yet (as of 2.0.0-dev.20160704).
```
$ tsc -v && tsc -w

Version 2.0.0-dev.20160704
src/controller/TestController.ts(15,3): error TS1311: Async functions are only available when targeting ECMAScript 2015 or higher.
src/controller/TestController.ts(15,9): error TS1057: An async function or method must have a valid awaitable return type.
```

**Comment by remojansen** [comment]
Looks like we will have to wait for async/await but I'm sure is really close I would use async/await and target ES6 for now. By the time typescript 2.0 is released we will be able to target ES5 with minimal changes.

**Comment by otbe** [comment]
Yes and as of now we can use target ```es6``` and use node v6 or ```--harmony``` where necessary. Pragmatic solution :)
